goodai/ltm/memory.py

```diff
@@ -282,3 +282,70 @@ def retrieve_from_keywords(self, keywords: list[str]) -> list[RetrievedMemory]:
                 metadata=chunk.metadata,
             ))
         return memories
+
+
+    def temporal_retrieval(self, origin_timestamp: float, temporal_hops: int) -> list[RetrievedMemory]:
+
+        mem: DefaultTextMemory = self.semantic_memory
+        search_index = 0
+        # Get the chunk we are searching from
+        for idx, chunk in enumerate(mem.get_all_chunks()):
+            if chunk.timestamp == origin_timestamp:
+                search_index = idx
+                break
+
+        temporal_chunks = self._timestamp_search(search_index, temporal_hops)
+
+        memories = []
+        for chunk in temporal_chunks:
+            text_key = chunk.associated_keys[0]
+            token_ids = mem.chunk_queue.get_sequence_token_ids(text_key)
+            text = mem.chunk_tokenizer.decode(token_ids, skip_special_tokens=True)
+
+            memories.append(RetrievedMemory(
+                passage=text,
+                passage_info=PassageInfo(
+                    chunk.from_token_seq_id, chunk.to_token_seq_id, token_ids,
+                ),
+                timestamp=chunk.timestamp,
+                distance=0.0,
+                relevance=0.0,
+                textKeys=[text_key],
+                metadata=chunk.metadata,
+            ))
```

Comment:
It looks like this could be extracted and reused in the keywords retrieval, to avoid code duplication.

Reply:
Maybe including the search among the chunks.

Reply:
Right. I forgot about those parts inbetween. We can leave it like it's now then.